checker/src/org/checkerframework/checker/index/samelen/SameLenVisitor.java

```diff
@@ -25,6 +25,23 @@ public SameLenVisitor(BaseTypeChecker checker) {
         super(checker);
     }
 
+    // This variable is used to store the name of the variable being assigned to on
+    // the left hand side of an assignment/pseudo-assignment, so that that variable
+    // can later be included in the SameLen type. This pattern is a bit uncomfortable,
+    // but there is not a clear other way to save this information.
```

Comment:
I tried this, and I couldn't quite get it to work. The problem I ran into is that to remove it you need to get the canonical name of the expression from the lhs' Tree. This is usually a `VariableTree`, but the `ExpressionTree` that I ought to get by calling `VariableTree#getNameExpression()` is always `null`, even though calling `getName()` on the same `VariableTree` results in a reasonable (but not canonical) name for the new variable. If `getNameExpression` actually returned an `ExpressionTree`, I could call `FlowExpressions.getInternalReprOf` to get a `Receiver`, which would have the canonical name if it was parsable to a String.